src/modules/qradar/aql_query_constructor.py

```diff
@@ -0,0 +1,143 @@
+                value = self._format_set(expression.value)
+            elif expression.comparator == ComparisonComparators.Equal or expression.comparator == ComparisonComparators.NotEqual:
+                # Should be in single-quotes
+                value = self._format_equality(expression.value)
+            # '%' -> '*' wildcard, '_' -> '?' single wildcard
+            elif expression.comparator == ComparisonComparators.Like:
+                value = self._format_like(expression.value)
+            else:
+                value = self._escape_value(expression.value)
+
+            comparison_string = ""
+            mapped_fields_count = len(mapped_fields_array)
+
+            for mapped_field in mapped_fields_array:
+                comparison_string += "{mapped_field}{comparator}{value}".format(
+                    mapped_field=mapped_field, comparator=comparator, value=value)
+                if (mapped_fields_count > 1):
+                    comparison_string += " OR "
+                    mapped_fields_count -= 1
+
+            if expression.comparator == ComparisonComparators.NotEqual:
+                comparison_string = self._negate_comparison(comparison_string)
+
+            if expression.negated:
+                comparison_string = self._negate_comparison(comparison_string)
+
+            return "{comparison}".format(comparison=comparison_string)
+
+        elif isinstance(expression, CombinedComparisonExpression):
+            query_string = "{} {} {}".format(self._parse_expression(expression.expr1),
+                                             self.comparator_lookup[expression.operator],
+                                             self._parse_expression(expression.expr2))
+            return query_string
+        elif isinstance(expression, ObservationExpression):
+            return self._parse_expression(expression.comparison_expression)
+        elif isinstance(expression, CombinedObservationExpression):
+            operator = self.comparator_lookup[expression.operator]
+            return "{expr1} {operator} {expr2}".format(expr1=self._parse_expression(expression.expr1),
+                                                       operator=operator,
+                                                       expr2=self._parse_expression(expression.expr2))
+        elif isinstance(expression, Pattern):
+            return "{opening_select} {expr}".format(opening_select=self.select_prefix, expr=self._parse_expression(expression.expression))
+        else:
+            raise RuntimeError("Unknown Recursion Case for expression={}, type(expression)={}".format(
+                expression, type(expression)))
+
+    def parse_expression(self, pattern: Pattern):
+        return self._parse_expression(pattern)
+
+
+def translate_pattern(pattern: Pattern, data_model_mapping):
+    x = AqlQueryStringPatternTranslator(pattern, data_model_mapping)
+    select_statement = "*"
```

Comment:
Can probably remove this line since it's being immediately overwritten just below